feat: add `stats/incr/nanprod`

jalajk3004 opened the pull request:
Resolves #5619

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package nanprod, which works like prod in which it calculates prod incrementally ignoring NaN values
- 
![image](https://github.com/user-attachments/assets/f6113428-eaa3-40d6-9705-d549f6d64d80)
![image](https://github.com/user-attachments/assets/e5a51ded-4d1c-4096-b6c2-c8773bd04bba)
![image](https://github.com/user-attachments/assets/4cc5d3b4-0938-41f5-ab6e-d58d9861203a)


## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5619

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

jalajk3004 committed:
edit

jalajk3004 committed:
edit

jalajk3004 committed:
Merge branch 'develop' of https://github.com/jalajk3004/stdlib into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
added nanprod

jalajk3004 committed:
lint

jalajk3004 committed:
check

jalajk3004 committed:
lint

jalajk3004 committed:
spaces

jalajk3004 committed:
check

jalajk3004 committed:
exanmpe

jalajk3004 committed:
exanmpe

jalajk3004 committed:
check

jalajk3004 committed:
check

jalajk3004 committed:
done

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into nanprod

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanprod/docs/types/index.d.ts:
```diff
@@ -0,0 +1,67 @@
+*
+* ## Notes
+*
+* -   If provided `NaN` or a value which, when used in computations, results in `NaN`, the accumulated value is `NaN` for all future invocations.
```
This note was copied from `prod` but is incorrect for `nanprod`. The whole purpose of this package is to *ignore* NaN values, not propagate them. The tests at lines 220-222 and 246-248 explicitly verify this behavior (e.g., `'ignores NaN and retains +infinity'`).

```suggestion
* -   If provided `NaN`, the value is ignored and the accumulated product remains unchanged.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanprod/package.json:
```diff
@@ -0,0 +1,65 @@
+{
+  "name": "@stdlib/stats/incr/nanprod",
+  "version": "0.0.0",
+  "description": "Compute a product incrementally while ignoring `NaN` values.",
```
No backticks around `NaN` in `package.json` descriptions.

```suggestion
  "description": "Compute a product incrementally while ignoring NaN values.",
```

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanprod/examples/index.js:
```diff
@@ -0,0 +1,40 @@
+for ( i = 0; i < 100; i++ ) {
+	v = ( bernoulli( 0.8 ) < 1 ) ? NaN : uniform( 0.0, 100.0 );
+	prod = accumulator( v );
+	console.log( '%d\t%d', v.toFixed( 4 ), ( prod === null ) ? NaN : prod.toFixed( 4 ) );
```
`.toFixed()` returns a string, so `%d` will coerce it back to a number and strip trailing zeros. Use `%s` instead.

```suggestion
	console.log( '%s\t%s', v.toFixed( 4 ), ( prod === null ) ? 'null' : prod.toFixed( 4 ) );
```

Planeshifter committed:
fix: address review feedback in stats/incr/nanprod

- Add the missing '// MAIN //' section header (lint requires it) and drop the Method/References JSDoc blocks copied from incr/prod, since this package delegates
- Remove the incorrect NaN-propagation note from index.d.ts (it ignores NaN)
- Drop backticks from package.json description and tidy README wording
- Fix examples output to use %s and guard against a null accumulated value
- Bump copyright year to 2026

Planeshifter reviewed (approved):
Thanks!
